### PR TITLE
[datadog_users] Added support for the Users data source

### PR DESCRIPTION
**Pull request by aldrickdev**
This PR adds the `users` data resource to the provider, this allows you to query for a list of users instead of just a single one.

### CODE REVIEW

File: docs/data-sources/users.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+### Optional
+
+- `filter` (String) Filter all users by the given string.
+- `filter_status` (String) Filter on status attribute. Comma separated list, with possible values Active, Pending, and Disabled
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+- `users` (List of Object) List of users. (see [below for nested schema](#nestedatt--users))
```
Comment (drichards-87) [inline docs/data-sources/users.md:33]:
```suggestion
- `users` (List of Object) List of users (see [below for nested schema](#nestedatt--users)).
```

File: docs/data-sources/users.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Optional
+
+- `filter` (String) Filter all users by the given string.
+- `filter_status` (String) Filter on status attribute. Comma separated list, with possible values Active, Pending, and Disabled
```
Comment (drichards-87) [inline docs/data-sources/users.md:28]:
```suggestion
- `filter_status` (String) Filter on status attribute. Comma-separated list with possible values of Active, Pending, and Disabled.
```

File: docs/data-sources/users.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "datadog_users Data Source - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Use this data source to retrieve information about existing users to use it in an other resources.
+---
+
+# datadog_users (Data Source)
+
+Use this data source to retrieve information about existing users to use it in an other resources.
```
Comment (drichards-87) [inline docs/data-sources/users.md:11]:
```suggestion
Use this data source to retrieve information about existing users for use in other resources.
```

File: docs/data-sources/users.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "datadog_users Data Source - terraform-provider-datadog"
+subcategory: ""
+description: |-
+  Use this data source to retrieve information about existing users to use it in an other resources.
```
Comment (drichards-87) [inline docs/data-sources/users.md:6]:
```suggestion
  Use this data source to retrieve information about existing users for use in other resources.
```

**Review by drichards-87 — APPROVED** [review]
Left a couple of suggestions from Docs and approved the PR.

**Review by skarimo — APPROVED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**